src/families/cosmos/libcore-postBuildAccount.js

```diff
@@ -0,0 +1,143 @@
+// @flow
+import type { Account } from "../../types";
+import type { CoreAccount } from "../../libcore/types";
+import type { CosmosResources, CoreCosmosLikeAccount } from "./types";
+import { BigNumber } from "bignumber.js";
+import { log } from "@ledgerhq/logs";
+import {
+  libcoreAmountToBigNumber,
+  libcoreBigIntToBigNumber,
+} from "../../libcore/buildBigNumber";
+import { promiseAllBatched } from "../../promise";
+
+const getValidatorStatus = async (
+  cosmosAccount: CoreCosmosLikeAccount,
+  address
+) => {
+  const status = ["unbonded", "unbonding", "bonded"];
+  const validatorInfo = await cosmosAccount.getValidatorInfo(address);
+  return status[await validatorInfo.getActiveStatus()];
+};
+
+const getFlattenDelegation = async (cosmosAccount) => {
+  const delegations = await cosmosAccount.getDelegations();
+
+  return await promiseAllBatched(10, delegations, async (delegation) => {
+    const validatorAddress = await delegation.getValidatorAddress();
+    const pendingRewards = await cosmosAccount.getPendingRewards();
```

Comment:
is there any reason why the pendingRewards fetching and the loop was not BEFORE the promiseAll? we are doing this again and again for each delegation, could it be a performance issue?

Reply:
Totally perf issue. I could've done in a smarter way 👍